assets/prototype/domain/eventEditor/containers/queries/useFetchRelations.js

```diff
@@ -0,0 +1,33 @@
+import { useQuery } from '@apollo/react-hooks';
+import { GET_RELATIONS_DATA } from './relations';
+import useToaster from '../../../../infrastructure/services/toaster/useToaster';
+
+const useFetchRelations = ({ eventId, onReceiveRelations }) => {
+	console.log('%c useFetchRelations: ', 'color: deeppink; font-size: 14px;');
+
+	const toaster = useToaster();
+	const toasterDatesMessage = 'initializing event editor relations';
+
+	const { data, error, loading } = useQuery(GET_RELATIONS_DATA, {
+		variables: { eventId },
+		onCompleted: (r_data) => {
+			if (onReceiveRelations) {
+				onReceiveRelations(r_data);
+			}
+			toaster.dismiss(toasterDatesMessage);
+			toaster.success(`event editor relations initialized`);
+		},
+		onError: (error) => {
+			toaster.dismiss(toasterDatesMessage);
+			toaster.error(error);
+		},
+	});
+
+	return {
+		data,
+		error,
+		loading,
```

Comment:
Not sure if we need to pass `error` and `loading` down.

Reply:
At least I am not sure if the consumer of this hook should be aware of these props.

Reply:
ya I think I'm leaning this way as well... at first I was thinking that it didn't hurt to pass these props down, but they are already being handled and you're right that the consumer doesn't need to be aware of them since it should only be concerned with the data and not be all tangled up with error handling, etc.

That said, `loading` could still be useful to the consumer.

Reply:
I made use of `onCompleted` callback to avoid usage of `useEffect` in relations manager because initially `data` is empty. So it needed many checks in `useEffect`. `onCompleted` is called only when it receives the data.

Reply:
???
not grokking how that pertains to returning `error` and `loading` from this hook

Reply:
Yeah that doesn't 😄 
Returned `error` and `loading` just in case a consumer might be interested in them, as rightly mentioned by you.